pghoard/rohmu/compressor.py

```diff
@@ -5,171 +5,27 @@
 See LICENSE for details
 """
 
-from pghoard.rohmu import IO_BLOCK_SIZE
-from pghoard.rohmu.encryptor import Encryptor, Decryptor, DecryptorFile
-from pghoard.rohmu.errors import InvalidConfigurationError, MissingLibraryError
-import logging
+from .snappyfile import SnappyFile
 import lzma
-import os
-import time
 
-try:
-    import snappy
-except ImportError:
-    snappy = None
 
+def CompressionFile(dst_fp, algorithm, level=0):
+    """This looks like a class to users, but is actually a function that instantiates a class based on algorithm."""
+    if algorithm == "lzma":
+        return lzma.open(dst_fp, "w", preset=level)
 
-class SnappyFile:
-    def __init__(self, fp):
-        self._comp = snappy.StreamDecompressor()
-        self._fp = fp
-        self._done = False
-        self._pos = 0
+    if algorithm == "snappy":
+        return SnappyFile(dst_fp, "wb")
 
-    def __enter__(self):
-        return self
+    return dst_fp
```

Comment:
good point, this logic was copied from the old implementation, but the new implementation can do something smarter about invalid values